diff --git a/dist/elasticui.js b/dist/elasticui.js
index 0afaa51..3402d7e 100644
--- a/dist/elasticui.js
+++ b/dist/elasticui.js
@@ -643,7 +643,10 @@ var elasticui;
                 });
                 $scope.$watch('indexVM.page', function () { return _this.search(); });
                 $scope.$watch('indexVM.index', function () { return _this.search(); });
-                $scope.$watch('indexVM.query', function () { return _this.search(); });
+                $scope.$watch('indexVM.query', function () {
+                    _this.indexVM.page = 1;
+                    return _this.search();
+                });
                 $scope.$watch('indexVM.highlight', function () { return _this.search(); });
                 $timeout(function () { return _this.loaded(); }, 200); // TODO: find better way to recognize loading of app
             }
