diff --git a/examples/demo/lib/elastic.js b/examples/demo/lib/elastic.js
index e52f20b..2b7ddc8 100644
--- a/examples/demo/lib/elastic.js
+++ b/examples/demo/lib/elastic.js
@@ -8922,6 +8922,26 @@
             },
 
             /**
+                  Allows you to set a specified filter on this query.
+
+                  @member ejs.BoolQuery
+                  @param {Object} filter Any valid <code>Filter</code> object.
+                  @returns {Object} returns <code>this</code> so that calls can be chained.
+                  */
+            filter: function (filter) {
+                if (filter == null) {
+                    return query.bool.filter;
+                }
+
+                if (!isFilter(filter)) {
+                    throw new TypeError('Argument must be a Filter');
+                }
+
+                query.bool.filter = filter.toJSON();
+                return this;
+            },
+
+            /**
                   <p>Sets the number of optional clauses that must match.</p>
             
                   <p>By default no optional clauses are necessary for a match
