--- plugins/destinations/file.inc	2012-10-31 21:55:13.000000000 +0000
+++ plugins/destinations/file.new	2014-10-03 16:58:07.018619266 +0000
@@ -243,7 +243,11 @@
     // replacing it
     if ($this->fileReplace == self::FILE_EXISTS_REUSE) {
       // See if we this file already (we'll reuse a file entity if it exists).
-      if (file_exists($destination)) {
+
+      // In order to speed up one-time file migrations, we are removing this
+      // check and always assuming that the file exists.
+      // if (file_exists($destination)) {
+      if (true) {
         $file = $this->createFileEntity($destination, $owner);
         // File entity didn't already exist, create it
         if (empty($file->fid)) {
