diff --git a/modules/system/system.install b/modules/system/system.install
index ae55b89..21b1592 100644
--- a/modules/system/system.install
+++ b/modules/system/system.install
@@ -583,10 +583,12 @@ function _system_check_db_utf8mb4_requirements($phase) {
       $requirements['severity'] = REQUIREMENT_INFO;
     }
     else {
-      // Not supported, not active.
-      $requirements['value'] = $t('Disabled');
-      $requirements['description'] = $t('4 byte UTF-8 for @driver is disabled. See the <a href="@url">documentation on adding 4 byte UTF-8 support</a> for more information.', array('@driver' => $driver, '@url' => $documentation_url));
-      $requirements['severity'] = REQUIREMENT_INFO;
+      if (variable_get('fa_display_utf8_4byte_drush_warning', FALSE)) {
+        // Not supported, not active.
+        $requirements['value'] = $t('Disabled');
+        $requirements['description'] = $t('4 byte UTF-8 for @driver is disabled. See the <a href="@url">documentation on adding 4 byte UTF-8 support</a> for more information.', array('@driver' => $driver, '@url' => $documentation_url));
+        $requirements['severity'] = REQUIREMENT_INFO;
+      }
     }
   }
   return $requirements;
