diff --git a/admin_views_default/node.admin-content.inc b/admin_views_default/node.admin-content.inc
index d501c18..977c540 100644
--- a/admin_views_default/node.admin-content.inc
+++ b/admin_views_default/node.admin-content.inc
@@ -129,128 +129,117 @@ $handler->display->display_options['fields']['views_bulk_operations']['element_l
 $handler->display->display_options['fields']['views_bulk_operations']['hide_alter_empty'] = FALSE;
 $handler->display->display_options['fields']['views_bulk_operations']['vbo_settings']['display_type'] = '0';
 $handler->display->display_options['fields']['views_bulk_operations']['vbo_settings']['enable_select_all_pages'] = 1;
+$handler->display->display_options['fields']['views_bulk_operations']['vbo_settings']['row_clickable'] = 1;
 $handler->display->display_options['fields']['views_bulk_operations']['vbo_settings']['force_single'] = 0;
 $handler->display->display_options['fields']['views_bulk_operations']['vbo_settings']['entity_load_capacity'] = '10';
 $handler->display->display_options['fields']['views_bulk_operations']['vbo_operations'] = array(
   'action::node_assign_owner_action' => array(
     'selected' => 1,
+    'postpone_processing' => 0,
     'skip_confirmation' => 1,
     'override_label' => 1,
     'label' => 'Change author',
-    'postpone_processing' => 0,
   ),
   'action::views_bulk_operations_delete_item' => array(
     'selected' => 1,
+    'postpone_processing' => 0,
     'skip_confirmation' => 0,
     'override_label' => 1,
     'label' => 'Delete',
+  ),
+  'action::node_make_sticky_action' => array(
+    'selected' => 1,
     'postpone_processing' => 0,
+    'skip_confirmation' => 1,
+    'override_label' => 1,
+    'label' => 'Stick',
+  ),
+  'action::node_make_unsticky_action' => array(
+    'selected' => 1,
+    'postpone_processing' => 0,
+    'skip_confirmation' => 1,
+    'override_label' => 1,
+    'label' => 'Unstick',
   ),
   'action::views_bulk_operations_modify_action' => array(
     'selected' => 1,
+    'postpone_processing' => 0,
     'skip_confirmation' => 1,
     'override_label' => 1,
     'label' => 'Change value',
     'settings' => array(
       'show_all_tokens' => 1,
       'display_values' => array(
-        '_all_' => '_all_',
+        'anthology::field_regions' => 'anthology::field_regions',
+        'anthology::field_topics' => 'anthology::field_topics',
+        'anthology::field_tags' => 'anthology::field_tags',
+        'article::field_regions' => 'article::field_regions',
+        'article::field_topics' => 'article::field_topics',
+        'article::field_tags' => 'article::field_tags',
+        'interview::field_regions' => 'interview::field_regions',
+        'interview::field_topics' => 'interview::field_topics',
+        'interview::field_tags' => 'interview::field_tags',
+        'listicle::field_regions' => 'listicle::field_regions',
+        'listicle::field_topics' => 'listicle::field_topics',
+        'listicle::field_tags' => 'listicle::field_tags',
+        'media::field_regions' => 'media::field_regions',
+        'media::field_topics' => 'media::field_topics',
+        'media::field_tags' => 'media::field_tags',
+        'news_release::field_regions' => 'news_release::field_regions',
+        'news_release::field_topics' => 'news_release::field_topics',
+        'news_release::field_tags' => 'news_release::field_tags',
+        'review::field_regions' => 'review::field_regions',
+        'review::field_topics' => 'review::field_topics',
+        'review::field_tags' => 'review::field_tags',
+        'sponsored_anthology::field_regions' => 'sponsored_anthology::field_regions',
+        'sponsored_anthology::field_tags' => 'sponsored_anthology::field_tags',
+        'sponsored_anthology::field_topics' => 'sponsored_anthology::field_topics',
       ),
     ),
-    'postpone_processing' => 0,
-  ),
-  'action::system_message_action' => array(
-    'selected' => 0,
-    'skip_confirmation' => 0,
-    'override_label' => 0,
-    'label' => '',
-    'postpone_processing' => 0,
-  ),
-  'action::views_bulk_operations_script_action' => array(
-    'selected' => 0,
-    'skip_confirmation' => 0,
-    'override_label' => 0,
-    'label' => '',
-    'postpone_processing' => 0,
   ),
-  'action::node_make_sticky_action' => array(
-    'selected' => 1,
-    'skip_confirmation' => 1,
-    'override_label' => 1,
-    'label' => 'Stick',
-    'postpone_processing' => 0,
-  ),
-  'action::node_make_unsticky_action' => array(
+  'action::views_bulk_operations_argument_selector_action' => array(
     'selected' => 1,
     'skip_confirmation' => 1,
     'override_label' => 1,
-    'label' => 'Unstick',
-    'postpone_processing' => 0,
-  ),
-  'action::views_bulk_operations_argument_selector_action' => array(
-    'selected' => 0,
-    'skip_confirmation' => 0,
-    'override_label' => 0,
-    'label' => '',
+    'label' => 'Create new Anthology',
     'settings' => array(
-      'url' => '',
+      'url' => 'fa-anthology/prepopulate/',
     ),
   ),
   'action::node_promote_action' => array(
     'selected' => 1,
+    'postpone_processing' => 0,
     'skip_confirmation' => 1,
     'override_label' => 1,
     'label' => 'Promote to front page',
-    'postpone_processing' => 0,
   ),
   'action::node_publish_action' => array(
     'selected' => 1,
+    'postpone_processing' => 0,
     'skip_confirmation' => 1,
     'override_label' => 1,
     'label' => 'Publish',
-    'postpone_processing' => 0,
   ),
-  'action::system_goto_action' => array(
-    'selected' => 0,
+  'action::fa_helper_reconcile_authors' => array(
+    'selected' => 1,
+    'postpone_processing' => 0,
     'skip_confirmation' => 0,
     'override_label' => 0,
     'label' => '',
-    'postpone_processing' => 0,
   ),
   'action::node_unpromote_action' => array(
     'selected' => 1,
+    'postpone_processing' => 0,
     'skip_confirmation' => 1,
     'override_label' => 1,
     'label' => 'Demote from front page',
-    'postpone_processing' => 0,
-  ),
-  'action::node_save_action' => array(
-    'selected' => 0,
-    'skip_confirmation' => 0,
-    'override_label' => 0,
-    'label' => '',
-    'postpone_processing' => 0,
-  ),
-  'action::system_send_email_action' => array(
-    'selected' => 0,
-    'skip_confirmation' => 0,
-    'override_label' => 0,
-    'label' => '',
-    'postpone_processing' => 0,
   ),
   'action::node_unpublish_action' => array(
     'selected' => 1,
+    'postpone_processing' => 0,
     'skip_confirmation' => 1,
     'override_label' => 1,
     'label' => 'Unpublish',
-    'postpone_processing' => 0,
-  ),
-  'action::node_unpublish_by_keyword_action' => array(
-    'selected' => 0,
-    'skip_confirmation' => 0,
-    'override_label' => 0,
-    'label' => '',
-    'postpone_processing' => 0,
   ),
 );
 /* Field: Content: Title */
@@ -274,6 +263,19 @@ $handler->display->display_options['fields']['name']['field'] = 'name';
 $handler->display->display_options['fields']['name']['relationship'] = 'uid';
 $handler->display->display_options['fields']['name']['label'] = 'Author';
 $handler->display->display_options['fields']['name']['hide_alter_empty'] = FALSE;
+/* Field: Content: Tags */
+$handler->display->display_options['fields']['field_tags']['id'] = 'field_tags';
+$handler->display->display_options['fields']['field_tags']['table'] = 'field_data_field_tags';
+$handler->display->display_options['fields']['field_tags']['field'] = 'field_tags';
+$handler->display->display_options['fields']['field_tags']['element_type'] = '0';
+$handler->display->display_options['fields']['field_tags']['element_label_colon'] = FALSE;
+$handler->display->display_options['fields']['field_tags']['element_wrapper_type'] = '0';
+$handler->display->display_options['fields']['field_tags']['element_default_classes'] = FALSE;
+$handler->display->display_options['fields']['field_tags']['settings'] = array(
+  'link' => 1,
+);
+$handler->display->display_options['fields']['field_tags']['delta_offset'] = '0';
+$handler->display->display_options['fields']['field_tags']['field_api_classes'] = TRUE;
 /* Field: Content: Published */
 $handler->display->display_options['fields']['status']['id'] = 'status';
 $handler->display->display_options['fields']['status']['table'] = 'node';
@@ -350,13 +352,126 @@ $handler->display->display_options['filters']['vid']['expose']['label'] = 'Vocab
 $handler->display->display_options['filters']['vid']['expose']['operator'] = 'vid_op';
 $handler->display->display_options['filters']['vid']['expose']['identifier'] = 'vid';
 $handler->display->display_options['filters']['vid']['expose']['remember'] = TRUE;
+/* Filter criterion: Content: Has taxonomy term */
+$handler->display->display_options['filters']['tid']['id'] = 'tid';
+$handler->display->display_options['filters']['tid']['table'] = 'taxonomy_index';
+$handler->display->display_options['filters']['tid']['field'] = 'tid';
+$handler->display->display_options['filters']['tid']['exposed'] = TRUE;
+$handler->display->display_options['filters']['tid']['expose']['operator_id'] = 'tid_op';
+$handler->display->display_options['filters']['tid']['expose']['label'] = 'Tags';
+$handler->display->display_options['filters']['tid']['expose']['operator'] = 'tid_op';
+$handler->display->display_options['filters']['tid']['expose']['identifier'] = 'tid';
+$handler->display->display_options['filters']['tid']['expose']['remember_roles'] = array(
+  2 => '2',
+  1 => 0,
+  7 => 0,
+  3 => 0,
+  4 => 0,
+  10 => 0,
+  13 => 0,
+  12 => 0,
+  8 => 0,
+  11 => 0,
+  6 => 0,
+  5 => 0,
+  9 => 0,
+  14 => 0,
+  15 => 0,
+  16 => 0,
+  17 => 0,
+  18 => 0,
+);
+$handler->display->display_options['filters']['tid']['vocabulary'] = 'tags';
 
 /* Display: System */
 $handler = $view->new_display('system', 'System', 'system_1');
 $handler->display->display_options['defaults']['hide_admin_links'] = FALSE;
 $handler->display->display_options['hide_admin_links'] = TRUE;
 $handler->display->display_options['defaults']['access'] = FALSE;
+$handler->display->display_options['defaults']['style_plugin'] = FALSE;
+$handler->display->display_options['style_plugin'] = 'table';
+$handler->display->display_options['style_options']['columns'] = array(
+  'views_bulk_operations' => 'views_bulk_operations',
+  'title' => 'title',
+  'timestamp' => 'title',
+  'type' => 'type',
+  'name' => 'name',
+  'field_tags' => 'field_tags',
+  'status' => 'status',
+  'changed' => 'changed',
+  'edit_node' => 'edit_node',
+  'delete_node' => 'edit_node',
+);
+$handler->display->display_options['style_options']['default'] = 'changed';
+$handler->display->display_options['style_options']['info'] = array(
+  'views_bulk_operations' => array(
+    'align' => '',
+    'separator' => '',
+    'empty_column' => 0,
+  ),
+  'title' => array(
+    'sortable' => 1,
+    'default_sort_order' => 'asc',
+    'align' => '',
+    'separator' => '&nbsp;',
+    'empty_column' => 0,
+  ),
+  'timestamp' => array(
+    'align' => '',
+    'separator' => '',
+    'empty_column' => 0,
+  ),
+  'type' => array(
+    'sortable' => 1,
+    'default_sort_order' => 'asc',
+    'align' => '',
+    'separator' => '',
+    'empty_column' => 0,
+  ),
+  'name' => array(
+    'sortable' => 1,
+    'default_sort_order' => 'asc',
+    'align' => '',
+    'separator' => '',
+    'empty_column' => 0,
+  ),
+  'field_tags' => array(
+    'align' => '',
+    'separator' => '',
+    'empty_column' => 0,
+  ),
+  'status' => array(
+    'sortable' => 1,
+    'default_sort_order' => 'desc',
+    'align' => '',
+    'separator' => '',
+    'empty_column' => 0,
+  ),
+  'changed' => array(
+    'sortable' => 1,
+    'default_sort_order' => 'desc',
+    'align' => '',
+    'separator' => '',
+    'empty_column' => 0,
+  ),
+  'edit_node' => array(
+    'align' => '',
+    'separator' => '&emsp;',
+    'empty_column' => 0,
+  ),
+  'delete_node' => array(
+    'align' => '',
+    'separator' => '',
+    'empty_column' => 0,
+  ),
+);
+$handler->display->display_options['style_options']['sticky'] = TRUE;
+$handler->display->display_options['style_options']['empty_table'] = TRUE;
+$handler->display->display_options['defaults']['style_options'] = FALSE;
+$handler->display->display_options['defaults']['row_plugin'] = FALSE;
+$handler->display->display_options['defaults']['row_options'] = FALSE;
 $handler->display->display_options['path'] = 'admin/content';
+
 $translatables['admin_views_node'] = array(
   t('Defaults'),
   t('Content'),
@@ -378,6 +493,7 @@ $translatables['admin_views_node'] = array(
   t('term'),
   t('- Choose an operation -'),
   t('Change author'),
+  t('Create new Anthology'),
   t('Delete'),
   t('Change value'),
   t('Stick'),
