diff --git a/modules/image/image.module b/modules/image/image.module
index 526330c..03afad0 100644
--- a/modules/image/image.module
+++ b/modules/image/image.module
@@ -1054,6 +1054,14 @@ function image_style_url($style_name, $path) {
     $file_url .= (strpos($file_url, '?') !== FALSE ? '&' : '?') . drupal_http_build_query($token_query);
   }
 
+  //Create derivated image if not exist
+  if(!file_exists($file_url) && !file_exists($uri)){
+    if(file_exists($path)){
+      $currentStyle = image_style_load($style_name);
+      image_style_create_derivative($currentStyle, $path, $uri);
+    }
+  }
+
   return $file_url;
 }
 
