diff --git a/includes/bootstrap.inc b/includes/bootstrap.inc
index b33f950..b525ad0 100644
--- a/includes/bootstrap.inc
+++ b/includes/bootstrap.inc
@@ -2950,6 +2950,11 @@ function ip_address() {
         // Eliminate all trusted IPs.
         $untrusted = array_diff($forwarded, $reverse_proxy_addresses);
 
+        // If configured, eliminate more trusted IPs.
+        if (variable_get('reverse_proxy_addresses_skip', 0) != 0) {
+          $untrusted = array_slice($untrusted, 0, -(variable_get('reverse_proxy_addresses_skip')));
+        }
+
         // The right-most IP is the most specific we can trust.
         $ip_address = array_pop($untrusted);
       }
