diff --git a/includes/bootstrap.inc b/includes/bootstrap.inc
index 7f1bfbad18..853d4433ab 100644
--- a/includes/bootstrap.inc
+++ b/includes/bootstrap.inc
@@ -3219,6 +3219,11 @@ function arg($index = NULL, $path = NULL) {
 function ip_address() {
   $ip_address = &drupal_static(__FUNCTION__);

+  // Using Imperva header to help resolve the IP address.
+  if (isset($_SERVER['HTTP_INCAP_CLIENT_IP']) && !empty($_SERVER['HTTP_INCAP_CLIENT_IP'])) {
+    return $_SERVER['HTTP_INCAP_CLIENT_IP'];
+  }
+
   if (!isset($ip_address)) {
     $ip_address = $_SERVER['REMOTE_ADDR'];

@@ -3241,6 +3246,11 @@ function ip_address() {
         // Eliminate all trusted IPs.
         $untrusted = array_diff($forwarded, $reverse_proxy_addresses);

+        // If configured, eliminate more trusted IPs.
+        if (!empty($untrusted) && variable_get('reverse_proxy_addresses_skip', 0) != 0) {
+          $untrusted = array_slice($untrusted, 0, -(variable_get('reverse_proxy_addresses_skip')));
+        }
+
         if (!empty($untrusted)) {
           // The right-most IP is the most specific we can trust.
           $ip_address = array_pop($untrusted);
