--- includes/file.inc	2014-10-15 10:36:46.000000000 -0400
+++ includes/file.inc.new	2014-10-20 15:55:22.000000000 -0400
@@ -598,14 +598,22 @@
  */
 function file_save(stdClass $file) {
   $file->timestamp = REQUEST_TIME;
-  $file->filesize = filesize($file->uri);
+
+  // This will get the filesize from S3 rather than trying to download the file
+  // and get the filesize from the local filesystem.
+  if (parse_url($file->uri)['scheme'] == 's3'){
+    $uri = preg_replace('/^s3:\/\//', 'http://'.variable_get('amazons3_bucket','').'.s3.amazonaws.com/', str_replace(' ','%20',$file->uri));
+    $ret = drupal_http_request($uri, array('method' => 'HEAD'));
+    $file->filesize = (int)$ret->headers['content-length'];
+  } else {
+    $file->filesize = filesize($file->uri);
+  }
 
   // Load the stored entity, if any.
   if (!empty($file->fid) && !isset($file->original)) {
     $file->original = entity_load_unchanged('file', $file->fid);
   }
 
-  module_invoke_all('file_presave', $file);
   module_invoke_all('entity_presave', $file, 'file');
 
   if (empty($file->fid)) {
