diff --git a/file.inc b/file.inc.new
index d3008cc..8bb3973 100644
--- a/file.inc
+++ b/file.inc.new
@@ -598,7 +598,16 @@ function file_load($fid) {
  */
 function file_save(stdClass $file) {
   $file->timestamp = REQUEST_TIME;
-  $file->filesize = filesize($file->uri);
+
+  // This will get the filesize from S3 rather than trying to download the file
+  // and get the filesize from the local filesystem.
+  if (parse_url($file->uri)['scheme'] == 's3'){
+    $uri = preg_replace('/^s3:\/\//', 'http://'.variable_get('amazons3_bucket','').'.s3.amazonaws.com/', str_replace(' ','%20',$file->uri));
+    $ret = drupal_http_request($uri, array('method' => 'HEAD'));
+    $file->filesize = (int)$ret->headers['content-length'];
+  } else {
+    $file->filesize = filesize($file->uri);
+  }
 
   // Load the stored entity, if any.
   if (!empty($file->fid) && !isset($file->original)) {
